enterprise/frontend/src/metabase-enterprise/comments/components/CommentEditor/editor.utils.ts

```diff
@@ -0,0 +1,42 @@
+import type { JSONContent } from "@tiptap/core";
+
+import type { DocumentContent } from "metabase-types/api";
+
+function isEmptyParagraphNode(node: JSONContent) {
+  if (node.type !== "paragraph") {
+    return false;
+  }
+  // Empty paragraph => no content or content has no text nodes with non-empty text
+  if (!node.content || node.content.length === 0) {
+    return true;
+  }
+  const hasNonEmptyText = node.content.some(
+    (child) =>
+      child.type === "text" &&
+      typeof child.text === "string" &&
+      child.text.trim().length > 0,
+  );
```

Comment:
This only checks for text nodes, we need to check for all types of nodes (inline links, inline mentions, etc.)

Reply:
b366a1d6fca4c729f14d070528c451e179904af1